### PR TITLE
[Pm-13097] Rename cryptoservice to keyservice and move it to km ownership

### PR DESCRIPTION
**Pull request by quexten**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-13097

## 📔 Objective

Since the crypto functions (en/decrypt symmetric and asymmetric) have been moved out of cryptoservice, it can now be moved to KM ownership and renamed to keyservice.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by dani-garcia — DISMISSED** [review]

File: apps/web/src/app/core/core.module.ts [hunk 1]
```diff
@@ -72,7 +71,7 @@ import {
   ThemeStateService,
 } from "@bitwarden/common/platform/theming/theme-state.service";
 import { VaultTimeout, VaultTimeoutStringType } from "@bitwarden/common/types/vault-timeout.type";
-import { BiometricsService } from "@bitwarden/key-management";
+import { KeyService as KeyServiceAbstraction, BiometricsService } from "@bitwarden/key-management";
```
Comment (mzieniukbw) [inline apps/web/src/app/core/core.module.ts:74]:
```suggestion
import { KeyService, BiometricsService } from "@bitwarden/key-management";
```

File: libs/common/src/auth/services/webauthn-login/webauthn-login-prf-crypto.service.ts
Comment (mzieniukbw) [inline libs/common/src/auth/services/webauthn-login/webauthn-login-prf-crypto.service.ts:1]: rename file from `webauthn-login-prf-crypto.service.ts` to `webauthn-login-prf-key.service.ts`

File: libs/common/src/auth/services/webauthn-login/webauthn-login-prf-crypto.service.spec.ts
Comment (mzieniukbw) [inline libs/common/src/auth/services/webauthn-login/webauthn-login-prf-crypto.service.spec.ts:1]: rename file from `webauthn-login-prf-crypto.service.spec.ts` to `webauthn-login-prf-key.service.spec.ts`

File: libs/common/src/auth/abstractions/webauthn/webauthn-login-prf-crypto.service.abstraction.ts
Comment (mzieniukbw) [inline libs/common/src/auth/abstractions/webauthn/webauthn-login-prf-crypto.service.abstraction.ts:1]: rename file from `webauthn-login-prf-crypto.service.abstraction.ts` to `webauthn-login-prf-key.service.abstraction.ts`

File: libs/tools/generator/core/src/engine/crypto-service-randomizer.ts
Comment (mzieniukbw) [inline libs/tools/generator/core/src/engine/crypto-service-randomizer.ts:1]: rename to `key-service-randomizer.ts`

File: libs/tools/generator/core/src/engine/crypto-service-randomizer.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
-/** A randomizer backed by a CryptoService. */
-export class CryptoServiceRandomizer implements Randomizer {
+/** A randomizer backed by a KeyService. */
+export class KeyServiceRandomizer implements Randomizer {
   /** instantiates the type.
    * @param crypto generates random numbers
    */
-  constructor(private crypto: CryptoService) {}
+  constructor(private crypto: KeyService) {}
```
Comment (mzieniukbw) [inline libs/tools/generator/core/src/engine/crypto-service-randomizer.ts:11]:
```suggestion
  constructor(private keyService: KeyService) {}
```

File: libs/tools/generator/core/src/engine/crypto-service-randomizer.spec.ts
Comment (mzieniukbw) [inline libs/tools/generator/core/src/engine/crypto-service-randomizer.spec.ts:1]: rename to `key-service-randomizer.spec.ts`

File: apps/web/src/app/core/core.module.ts (hunk 1 above)
Comment (mzieniukbw) [reply apps/web/src/app/core/core.module.ts:74]: Did not realise we do this rename with `Abstraction` suffix in all `*.module.ts`. Let's leave it consistent, resolving this comment.

**Review by jrmccannon — DISMISSED** [review]

**Review by mzieniukbw — APPROVED** [review]

File: libs/common/src/auth/services/webauthn-login/webauthn-login-prf-crypto.service.ts
Comment (quexten) [reply libs/common/src/auth/services/webauthn-login/webauthn-login-prf-crypto.service.ts:1]: Nice, thank you for catching these, must have missed them after the first find&replace sweep.

**Review by mzieniukbw — DISMISSED** [review]

**Review by dani-garcia — DISMISSED** [review]

**Review by dani-garcia — APPROVED** [review]

**Review by vleague2 — APPROVED** [review]

**Review by jrmccannon — DISMISSED** [review]

**Review by ike-kottlowski — APPROVED** [review]

**Review by mzieniukbw — APPROVED** [review]

**Review by djsmith85 — DISMISSED** [review]

**Review by amorask-bitwarden — DISMISSED** [review]

**Review by dani-garcia — APPROVED** [review]

**Review by djsmith85 — APPROVED** [review]

**Review by jrmccannon — APPROVED** [review]

**Review by vleague2 — APPROVED** [review]

**Review by amorask-bitwarden — APPROVED** [review]

**Review by ike-kottlowski — APPROVED** [review]

**Pull request merged**